### DISCUSSION TITLE
Immunedeconv CIBERSORT p value and permutations

### DISCUSSION THREAD

**Discussion by dy001-lee** [omnideconv/immunedeconv#173] [Q&A] [answered]
Please briefly describe your problem and what output you expect. If you have a question, please don't use this form.
Instead, ask on our [Community Forum](https://github.com/icbi-lab/immunedeconv/discussions)

Please include a minimal reproducible example (AKA a reprex). If you've never heard of a [reprex](http://reprex.tidyverse.org/) before, start by reading <https://www.tidyverse.org/help/#reprex>.

---

### Brief description of the problem
Hello,

I'm doing some deconvolution analysis with Immunedeconv package embedded function for CIBERSORT deconvolution. The source code and LM22 has been extracted from CIBERSORT itself. However, I realised the results from Immunedeconv gives me P value 9999 and there is no parameter function to annotate desired permutation under immunedeconv (apart from array, absolute and absolute_method). The p value results also vary when I try to run using web server for CIBERSORT.

Any ideas why or able to reproduce the same results, but using local R server?

Thanks!
```r
# insert reprex here
```

#### Versions

<details>

[Paste the output of `sessionInfo()` leaving a blank line after the details tag]

</details>

**Comment by grst** [comment, marked as answer]
Both `deconvolute` and `deconvolute_cibersort` use the `...` parameter, so you should be able to pass arbitrary arguments to the original CIBERSORT function, including something like `perm=1000`. 

That said, I believe that it's a more valid approach to derive p-values by comparing immune cell fractions between two groups of samples, e.g. using a linear model.
